mod_app/admin/bibliography_admin.py

```diff
@@ -37,3 +40,78 @@ def safe_annotation(self, obj):
     safe_annotation.allow_tags = True
     safe_annotation.short_description = "Annotations"
     safe_annotation.admin_order_field = "annotation"
+
+    # overiding django admin's changelist_view
+    def changelist_view(self, request):
+        if request.method == "POST" and request.FILES.get("html-file"):
+            html_file = request.FILES["html-file"]
+            if not html_file.name.endswith(".html") or html_file.name.endswith(".htm"):
+                self.message_user(
+                    request, "This is not a valid html file", level=messages.ERROR
+                )
+                return redirect(request.path)
+            try:
+                created_count, skipped_count = import_from_html(html_file)
+            except Exception as e:
+                self.message_user(request, e, level=messages.ERROR)
+                return redirect(request.path)
+
+            self.message_user(
+                request,
+                f"{created_count} bibliography items successfully created.",
+                level=messages.SUCCESS,
+            )
+            if skipped_count:
+                self.message_user(
+                    request,
+                    f"{skipped_count} items not created because they already exist",
+                    level=messages.WARNING,
+                )
+            return redirect(request.path)
+
+        return super().changelist_view(request)
+
+
+def import_from_html(html_file):
+    """Extract and create bibliography items from html table"""
+
+    created_count = 0
+    skipped_count = 0
+
+    soup = BeautifulSoup(html_file, "html.parser")
+    table = soup.find("table")
+    rows = table.find_all("tr")
+    for row in rows[1:]:
+        cols = row.find_all("td")
+        if len(cols) != 2:
+            raise ValueError()
+
+        # content duplication check
+        stripped_existing_short_citations = [
+            strip_tags(bib.short_citation) for bib in BibliographyItem.objects.all()
+        ]
+        stripped_existing_full_citations = [
+            strip_tags(bib.full_citation) for bib in BibliographyItem.objects.all()
+        ]
+        stripped_current_bibliography = [col.get_text(strip=True) for col in cols]
+        is_short = stripped_current_bibliography[0] in stripped_existing_short_citations
+        is_full = stripped_current_bibliography[1] in stripped_existing_full_citations
+
+        if is_short or is_full:
```

Comment:
we probably don't actually want to not import it if it's the same short citation, so just the full one should be fine